scapy/contrib/igmpv3.py

```diff
@@ -66,6 +75,59 @@ def mysummary(self):
     return self.sprintf("IGMPv3 Group Record %IGMPv3gr.type% %IGMPv3gr.maddr%")
 
 
+class IGMPv3Report(Packet):
+  """IGMP Report Class for v3.
+
+  The IGMP v3 Report header differs significantly from other IGMP messages,
+  therefore an specific class is implemented for it.
+
+  'chksum' is automatically calculated before the packet is sent.
+  """
+  name = "IGMPv3Report"
+  fields_desc = [XByteField("type", 0x22),  # Type 0x22 : "V3 Member Report"
+                 ByteField("resv", 0),
+                 XShortField("chksum", None),
+                 ShortField("resv2", 0),
+                 ShortField("numgrprecs", 0)]  # Update manually GR are added
+
+  def igmpize(self, ip=None, ether=None):  # noqa
```

Comment:
Scapy already does the job of computing the correct multicast MAC address, see [this line](https://github.com/secdev/scapy/blob/master/scapy/layers/l2.py#L56). 

Could you use `overload_fields` in IGMPv3Report() and check if it works as expected ?